app/src/main/java/ch/hikemate/app/model/profile/ProfileViewModel.kt

```diff
@@ -15,19 +15,19 @@ import kotlinx.coroutines.flow.asStateFlow
  * @param repository The profile repository.
  */
 open class ProfileViewModel(private val repository: ProfileRepository) : ViewModel() {
-  // The profile, i.e the profile for the detail view
-  // This will probably always be the current user profile
+
+  // The profile for the detail view
   private val profile_ = MutableStateFlow<Profile?>(null)
   val profile: StateFlow<Profile?> = profile_.asStateFlow()
 
   init {
-    repository.init {
-      // Get the profile of the current user
-      FirebaseAuth.getInstance().currentUser?.uid?.let { getProfileById(it) }
+    FirebaseAuth.getInstance().addAuthStateListener { auth ->
+      auth.currentUser?.uid?.let { userId -> getProfileById(userId) }
     }
+    repository.init { FirebaseAuth.getInstance().currentUser?.uid?.let { getProfileById(it) } }
   }
```

Comment:
Question: I'm not sure how this works. It looks to me that you are doing the job twice. Could you explain it ?

Reply:
The way I understand it, the code is doing two things:

1. It sets up an authentication state listener that fetches the user's profile whenever their authentication state changes. This ensures the profile is up-to-date as the user's login status evolves.
2. It also fetches the profile immediately when the code is first executed, without waiting for an authentication state change. This guarantees the profile is available right away, rather than waiting for the first state change event.
If for you it seems reasonable to remove the second I can do that.

Reply:
I think, if it works, you can try to remove it.

Reply:
I remove it then